src/BookNavigator/book-navigator.js

```diff
@@ -94,6 +92,31 @@ export class BookNavigator extends LitElement {
     }));
   }
 
+  /**
+   *  @typedef {{
+   *  baseHost: string,
+   *  modal: ModalManager,
+   *  sharedObserver: SharedResizeObserver,
+   *  bookreader: BookReader,
+   *  item: Item,
+   *  signedIn: boolean,
+   *  isAdmin: boolean,
+   *  onProviderChange: function,
```

Comment:
For this `function` definition, in TS, this would be
```suggestion
   *  onProviderChange: (BookReader, object) => void,
```
I'm not totally sure what the params passed back are, but you can define what that `function` is like this.

Reply:
thanks for catching. updated locally